src/Download.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react'
+import axios from 'axios'
+import fileDownload from 'js-file-download'
+
+export default function Download({ query }) {
+  const [selected, setSelected] = useState('json')
+  const options = ['json', 'csv', 'xlsx', 'tsv', 'ods']
+  const [downloading, setDownloading] = useState('')
+
+  const handleChange = (event) => {
+    setSelected(event.target.value)
+  }
+
+  const downloadData = () => {
+    setDownloading('Preparing Download')
+    axios(`http://localhost:8080/v1/download?format=${selected}`, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      responseType: 'blob',
+      data: {
+        query: `query Dataset {
+                    ${query}
+                    }
+                  `,
+      },
+    })
+      .then((response) => {
+        setDownloading('Downloading file')
```

Comment:
I think this is very basic version of a spinner but could we have a proper spinner instead? Also, I think this triggers after the response came back...

Reply:
We can use a proper spinner... I chose to add this because the file is downloaded as a variable and only after this the download is displayed. If the info is big enough or the user connection is slow, I think it a better give to the user a proper response while the file is being downloaded.